rpc/src/lib.rs

```diff
@@ -80,16 +88,26 @@ fn rich_block_build(block: ethereum::Block) -> RichBlock {
 			number: Some(block.header.number),
 			gas_used: block.header.gas_used,
 			gas_limit: block.header.gas_limit,
-			extra_data: Bytes(vec![]), // TODO H256 to Vec<u8>
+			extra_data: Bytes(block.header.extra_data.as_bytes().to_vec()),
 			logs_bloom: Some(block.header.logs_bloom),
 			timestamp: U256::from(block.header.timestamp),
 			difficulty: block.header.difficulty,
 			total_difficulty: None, // TODO
-			seal_fields: vec![], // TODO
+			seal_fields: vec![
+				Bytes(block.header.mix_hash.as_bytes().to_vec()),
+				Bytes(block.header.nonce.as_bytes().to_vec())
+			],
 			uncles: vec![], // TODO
-			// TODO expected struct `frontier_rpc_core::types::transaction::Transaction`,
-			// found struct `ethereum::transaction::Transaction`
-			transactions: BlockTransactions::Full(vec![]),
+			transactions: BlockTransactions::Full(
+				block.transactions.iter().enumerate().map(|(index,transaction)|{
+					let mut status = statuses[index].clone();
```

Comment:
Overall it looks good. I wonder if this is safe (and future proof). What if optimize the code to make the status retrieval parallel, they could come back in different order. (That was the reason I passed them by tuple (transaction, status)). We will fix the day it happens, if it happens :)

Reply:
Yes, definitely something to be aware of. If the transaction execution would happen in an offchain task, or some sort of async operation this approach would be incorrect.